### PR TITLE
[flutter_tools] Reland: use correct sdk path for analysis

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Reland of https://github.com/flutter/flutter/commit/9ed3e5bea6c36d4b4a25ebed26e4db8ebeee04f8 with fixed tests

## Description

flutter analyze was unconditionally using the cached dart-sdk for analysis, and was not running with the built SDK during local engine. This broke when trying to update the analyzer for null safety, since it required us to wait for the dart change to roll into the framework first.

### CODE REVIEW

File: packages/flutter_tools/test/commands.shard/hermetic/analyze_continuously_test.dart [hunk 1]
```diff
@@ -187,4 +196,88 @@ void main() {
...
+
+  testUsingContext('Can run AnalysisService with customized cache location', () async {
+    final Completer<void> completer = Completer<void>();
+    final StreamController<List<int>> stdin = StreamController<List<int>>();
+    final FakeProcessManager processManager = FakeProcessManager.list(
+      <FakeCommand>[
+        FakeCommand(
+          command: const <String>[
+            'custom-dart-sdk/bin/dart',
+            '--disable-dart-dev',
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/commands.shard/hermetic/analyze_continuously_test.dart:208]: this is the fix

File: packages/flutter_tools/test/commands.shard/hermetic/analyze_continuously_test.dart [hunk 2]
```diff
@@ -187,4 +196,88 @@ void main() {
...
+
+  testUsingContext('Can run AnalysisService with customized cache location --watch', () async {
+    final Completer<void> completer = Completer<void>();
+    final StreamController<List<int>> stdin = StreamController<List<int>>();
+    final FakeProcessManager processManager = FakeProcessManager.list(
+      <FakeCommand>[
+        FakeCommand(
+          command: const <String>[
+            'custom-dart-sdk/bin/dart',
+            '--disable-dart-dev',
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/commands.shard/hermetic/analyze_continuously_test.dart:249]: and this

**Review by jmagman — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
Landing to kick build

**Pull request merged**